operator/pkg/apis/core/v1beta1/configconnectorcontext_types.go

```diff
@@ -42,6 +42,18 @@ type ConfigConnectorContextSpec struct {
 	// Specifies the project to use for preconditions, quota and billing.
 	// Should only be used when requestProjectPolicy is set to BILLING_PROJECT.
 	BillingProject string `json:"billingProject,omitempty"`
+
+	// StateIntoSpec is the user override of the default value for the
+	// cnrm.cloud.google.com/state-into-spec annotation if the annotation is
+	// unset in a resource.
+	// 'absent' means unspecified fields in the resource spec stay unspecified
+	// after the successful reconciliations.
+	// 'merge' means the unspecified fields in the resource spec are populated
+	// after the successful reconciliations if the fields are computed/defaulted
+	// by the API.
+	//+kubebuilder:validation:Enum=absent;merge
+	//+kubebuilder:validation:Optional
+	StateIntoSpec *string `json:"stateIntoSpec,omitempty"`
```

Comment:
unless I misunderstood, Justin mentioned we may want to use upper case here  for the values themselves (Absent, Merge)

xref https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1300#discussion_r1505217530

If you define a default along the type, then maybe we don't need the SystemDefaultValue validation?

Reply:
@gemmahou @jingyih do you have any thoughts on the enum values we should use here?

Reply:
I might miss something but I'm wondering why the values are different at namespace/cluster level and resource level? Can we make them all upper case or lower case? While, I feel for resource annotations(like deletion-policy, etc) we always use lower case.

Reply:
> If you define a default along the type, then maybe we don't need the SystemDefaultValue validation?

@acpana sorry I forgot this question. What I want to achieve here is to make it clear to KCC whether the field is unset. Defining the default value in the type will make userOverride always populated in CCC when we get it in the controllers so the user intention can be vague. Knowing whether the field is set or not may help us determine proper warnings/instructions when we gradually "turn off" state-into-spec.

> I might miss something but I'm wondering why the values are different at namespace/cluster level and resource level? Can we make them all upper case or lower case?

@gemmahou  Currently the values at namespace/cluster level and resource level are not different, they are all lowercase values. We should not use the values with the first letter capitalized for resource-level configuration because this is an existing feature, and changing the available values is kinda expensive. If consistency is more important, then we should make all of them (namespace/cluster level and resource level) in lowercase.

Reply:
KRM Enums are upper-case: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants

We can support both values if we think it will cause confusion, I don't think it will.

Reply:
SG. We should keep an eye on the customer feedback about it.